pkg/importer/dataStream.go

```diff
@@ -485,24 +400,46 @@ func CopyData(dso *DataStreamOptions) error {
 		}
 		return nil
 	}
-	return ds.copy(dso.Dest)
+	requiresScratch, err := ds.copy(dso.Dest)
+	dso.RequiresScratch = requiresScratch
+	return err
 }
 
 // SaveStream reads from a stream and saves data to dest
-func SaveStream(stream io.ReadCloser, dest string) (int64, error) {
+func SaveStream(stream io.ReadCloser, dest string, diskImageFileName, dataPath, scratchPath string) (int64, error) {
 	glog.V(1).Infof("Saving stream to %q...\n", dest)
-	ds, err := newDataStreamFromStream(stream)
+	ds, err := newDataStream(&DataStreamOptions{
```

Comment:
I know you moved this from elsewhere, but how about using property name constructor or whatever it's called